pkg/index/index.go

```diff
@@ -156,7 +157,8 @@ type Writer interface {
 
 // FieldIterable allows building a FieldIterator.
 type FieldIterable interface {
-	Iterator(fieldKey FieldKey, termRange RangeOpts, order modelv1.Sort, preLoadSize int) (iter FieldIterator[*DocumentResult], err error)
+	Query(seriesMatchers []SeriesMatcher, secondaryQuery Query) (Query, error)
+	Iterator(fieldKey FieldKey, termRange RangeOpts, order modelv1.Sort, preLoadSize int, query Query, fields []string) (iter FieldIterator[*DocumentResult], err error)
```

Comment:
```suggestion
	Iterator(fieldKey FieldKey, termRange RangeOpts, order modelv1.Sort, preLoadSize int, query Query, fields []FieldKey) (iter FieldIterator[*DocumentResult], err error)
```

Please ensure that the semantics are consistent with `SeriesStore.Search`.